### PR TITLE
uric/contact: fix display name and contact header uri escaping

### PR DESCRIPTION
**Pull request by maximilianfridrich**
In re, display names are always sent as quoted-string which must not be escaped. Further, basically all UTF-8 characters are allowed in quoted-strings, so not checking it should be fine for almost all cases. The relevant rules in the BNF of RFC 3261 section 25.1:

```
      UTF8-NONASCII   =  %xC0-DF 1UTF8-CONT
                      /  %xE0-EF 2UTF8-CONT
                      /  %xF0-F7 3UTF8-CONT
                      /  %xF8-Fb 4UTF8-CONT
                      /  %xFC-FD 5UTF8-CONT
      UTF8-CONT       =  %x80-BF

      quoted-string  =  SWS DQUOTE *(qdtext / quoted-pair ) DQUOTE
      qdtext         =  LWS / %x21 / %x23-5B / %x5D-7E
                        / UTF8-NONASCII
      quoted-pair  =  "\" (%x00-09 / %x0B-0C
                      / %x0E-7F)

      display-name   =  *(token LWS)/ quoted-string
```

So the special character 0x00-0x20 (execpt LWS), 0x7F, '\', '"', CR and LF are not allowed. And the `quoted-pair` rules even allows `'"'` and `'\'` (and any other ASCII character except for CR and LF) as long as they are preceded by a `'\'`. 

Also, SIP URIs in the Contact header must be escaped.

### CODE REVIEW

**Comment by cspiel1** [comment]
Looks good to me.

Escaping URI  parameters will be handled in a separate PR.

**Comment by cspiel1** [comment]
Then we should remove `uri_display_name_escape()` completely. It is currently not used.

**Comment by cspiel1** [comment]
We shortly added this function. So most likely nobody depends on it.
The PR looks good for me.

File: vgcore.51007
Comment (sreimers) [inline vgcore.51007:1]: looks like this file was added by mistake

Comment (maximilianfridrich) [reply vgcore.51007:1]: Thanks. Fixed.

**Comment by alfredh** [comment]
I think this one looks ready for merge ...

**Pull request merged**